common/src/main/java/com/sequenceiq/cloudbreak/logger/MDCBuilder.java

```diff
@@ -124,6 +124,31 @@ public static void buildMdcContext(Object object) {
         getOrGenerateRequestId();
     }
 
+
+    public static void buildMdcContextFromInfoProvider(MdcContextInfoProvider object) {
+        if (object == null) {
+            MDC.put(LoggerContextKey.USER_CRN.toString(), null);
+            MDC.put(LoggerContextKey.RESOURCE_TYPE.toString(), null);
+            MDC.put(LoggerContextKey.RESOURCE_ID.toString(), null);
+            MDC.put(LoggerContextKey.RESOURCE_CRN.toString(), null);
+            MDC.put(LoggerContextKey.RESOURCE_NAME.toString(), null);
+        } else {
+            MDC.put(LoggerContextKey.WORKSPACE.toString(), object.getWorkspaceName());
+            MdcContext.builder()
+                    .resourceCrn(object.getResourceCrn())
+                    .resourceType(object.getResourceType())
+                    .resourceName(object.getResourceName())
+                    .environmentCrn(object.getEnvironmentCrn())
+                    .tenant(object.getTenantName())
+                    .buildMdc();
+        }
+        getOrGenerateRequestId();
+    }
+
+    public static void buildMdcContext(MdcContextInfoProvider object) {
```

Comment:
It is useful. We try to determine the fields value with reflection. We are changing the POJO to interfaces and the interface does not have the required fields it will always null. Now, the interfaces says the required values instead of the reflection magic

Reply:
as discussed it is to help using the `buildMdcContext` method in general without the need to change many calls to the `buildMdcContextFromInfoProvider`